core/project_settings.h

```diff
@@ -156,6 +158,14 @@ class ProjectSettings : public Object {
 
 	bool has_custom_feature(const String &p_feature) const;
 
+	// Either use the signal `project_settings_changed` or query this function.
+	// N.B. This will return true for TWO frames rather than just one. This is to cope
+	// the situation where a project setting is changed in the iteration AFTER it is read.
+	// There is therefore the potential for a change to be missed. Persisting the flag
+	// for two frames avoids this, at the cost of a double read.
+	bool is_dirty_this_frame() const { return _dirty_this_frame != 0; }
```

Comment:
Note that this could also potentially be set to:
```
	bool is_dirty_this_frame() const { return _dirty_this_frame == 1; }
```
In order to ensure the change only happens once, at a cost of a frame delay. We can tweak this quite easily if necessary, or I can change here if desired.